### PR TITLE
Clean up client constructor and how we use attr_reader/writer

### PR DESCRIPTION
**Pull request by hfaulds**
It's been bugging me that we have such a confusing constructor.

Some of it got moved to `Rets::HttpClient`

I tried to clean up where we refer to instance variables with attr_readers within the class (sometimes we would refer to them as `self.foo`,`@foo` or `foo`).

I removed the attr_writer and allowed those arguments to be passed into the constructor.

The only instance var I left that confuses me is `@tries` which appears to be some now-defunct code which I would be in favour of deleting.

### CODE REVIEW

File: lib/rets/http_client.rb [hunk 1]
```diff
@@ -10,6 +10,35 @@ def initialize(http, options, logger, login_url)
...
+      end
+
+      http_client = new(http, options, logger, options[:login_url])
+
+      if options[:http_timing_stats_collector]
+        http_client = Rets::MeasuringHttpClient.new(http_client, options.fetch(:http_timing_stats_collector), options.fetch(:http_timing_stats_prefix))
+      end
+      if options[:lock_around_http_requests]
+        http_client = Rets::LockingHttpClient.new(http_client, options.fetch(:locker), options.fetch(:lock_name), options.fetch(:lock_options))
+      end
```
Comment (phiggins) [inline lib/rets/http_client.rb]: It seems a bit weird that you get either `Rets::MeasuringHttpClient` or `Rets::LockingHttpClient`. Could they compose (ie, one wrapping the other)? Is there no use case where you might want both?

Comment (phiggins) [inline lib/rets/http_client.rb]: Oh I guess they do, duh. Reading comprehension failure on my part.

Comment (tcrayford) [inline lib/rets/http_client.rb]: apologese

Comment (hfaulds) [inline lib/rets/http_client.rb]: I added some whitespace to make it easier to comprehend

**Pull request merged**